railties/lib/rails/generators/rails/plugin/plugin_generator.rb

```diff
@@ -51,7 +52,7 @@ def readme
     end
 
     def gemfile
-      template "Gemfile"
+      template "Gemfile" unless inside_application?
```

Comment:
> Are we skipping the `Gemfile` because we're using the app as a test dummy, and thus we need to bring in it's dependencies? That seems unfortunate somehow.

Yeah that's pretty much it. I agree bringing in the dependencies of the whole app is probably not ideal (especially if the goal is testing a single plugin-in-application), so I may need to think about this more.

> We could possibly use [`eval_gemfile`](https://www.rubydoc.info/gems/bundler/Bundler%2FDsl:eval_gemfile) to include the app's `Gemfile` in the plugin's `Gemfile`, but I'm not sure if that works with a `gemspec` call in both `Gemfile`.

I'm not understanding how this would help, at that point what's the benefit of having separate `Gemfile`s?

Reply:
> I'm not understanding how this would help, at that point what's the benefit of having separate `Gemfile`s?

A dedicated `Gemfile` would give users a place to add development dependencies that are specific to the plugin.  Also, similar to `.gitignore`, if the plugin is eventually extracted as a separate gem, a `Gemfile` would be one less file for the user to have to create.  (Though they would have to edit it slightly.)